nutanix/data_source_nutanix_era_clusters.go

```diff
@@ -0,0 +1,252 @@
+							Computed: true,
+						},
+						"status": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"version": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"hypervisor_type": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"hypervisor_version": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"properties": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"ref_id": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"name": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"value": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"secure": {
+										Type:     schema.TypeBool,
+										Computed: true,
+									},
+									"description": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+								},
+							},
+						},
+						"reference_count": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"username": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"password": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"cloud_info": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"resource_config": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"storage_threshold_percentage": {
+										Type:     schema.TypeFloat,
+										Computed: true,
+									},
+									"memory_threshold_percentage": {
+										Type:     schema.TypeFloat,
+										Computed: true,
+									},
+								},
+							},
+						},
+						"management_server_info": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"entity_counts": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"healthy": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceNutanixEraClustersRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*Client).Era
+
+	resp, err := conn.Service.ListClusters(ctx)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if e := d.Set("clusters", flattenClustersResponse(resp)); err != nil {
+		return diag.FromErr(e)
+	}
+
+	uuid, er := uuid.GenerateUUID()
+
+	if er != nil {
+		return diag.Errorf("Error generating UUID for era clusters: %+v", err)
+	}
+	d.SetId(uuid)
+	return nil
+}
+
+func flattenClustersResponse(crsp *Era.ClusterListResponse) []map[string]interface{} {
+	if crsp != nil {
+		lst := []map[string]interface{}{}
+		for _, v := range *crsp {
+			d := map[string]interface{}{}
```

Comment:
Shouldn't we reuse the setter code of the singular datasource here ?

Reply:
singular datasource are directly set w/o any method.